### PR TITLE
feat(sdk): show jira posture layering on the graph

### PR DESCRIPTION
**Pull request by jonathanhaaswriter**
## Summary
- add SDK helpers for bounded graph neighborhood reads in both Python and TypeScript
- extend the Jira posture onboarding examples to fetch graph neighborhoods after claim writes so the layered workspace/project topology is visible immediately
- return graph-layering output alongside submitted and persisted claims, with graceful API error payloads when the graph surface is unavailable

## Testing
- make verify
- python3 -m py_compile sdk/python/cerebro_sdk/client.py sdk/python/examples/jira_posture_onboarding.py
- PYTHONPATH=sdk/python python3 smoke test for graph helpers and build_workspace_claims
- cd sdk/typescript && npx -y -p typescript tsc -p tsconfig.json
- cd sdk/typescript && npx -y -p tsx tsx smoke test for graph helpers and buildWorkspaceClaims

### CODE REVIEW

**Comment by jonathanhaaswriter** [comment]
@droid Please perform a deep code review and security review of this PR. Leave inline comments only for high-confidence, actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review this PR after the latest fixes. Focus on whether the prior code review and security findings are resolved, and call out any remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
### Droid sweep status

Latest sweep: `2026-04-29 22:09 UTC`.

- Unresolved non-outdated review threads: `0`
- CI `verify`: `SUCCESS`
- `droid-review`: `not present`

I’ll keep watching for new review findings/check failures and address them as they appear.

**Comment by jonathanhaaswriter** [comment]
Re-reviewed all live unresolved review threads on the current branch tip, including outdated threads. The current tip already addresses them: Python graph helpers now validate/normalize root URNs, `GraphEntity.label` is optional in the TypeScript SDK, and the unreachable empty-root branch in `graphLayering` is gone, so I resolved the threads. No code changes were needed, so I did not rerun tests or `make verify`. Remaining audited unresolved threads: none.

**Comment by jonathanhaaswriter** [comment]
Droid sweep audit complete (final pass)

- Re-scanned all live review threads, including outdated ones
- Verified each remaining thread against the current branch tip
- Resolved every actionable thread (or fixed it inline and pushed)
- 0 unresolved review threads across the whole open-PR stack at this time

PR #494 (fix/feedback-main-followup) and PR #495 (feat/droid-creation-workflows) received additional fixes for SSRF (DNS-rebind) and workflow branch-name validation in this pass.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
@droid please harden this PR end-to-end:

1. Re-review the diff for residual bugs, races, edge cases, security gaps, and untested paths.
2. Apply the smallest correct fixes for any high-confidence issues you find.
3. Strengthen test coverage: add regression tests for every prior review finding on this PR plus the new behavior, including failure paths, boundary inputs, concurrency, and not-found / typed-nil interface guards.
4. Run focused tests and `make verify`. If validators fail, fix until green.
5. Commit and push the fixes to this PR's branch (or, if you lack push permission in this workflow, leave concrete inline suggestions and a summary comment).

Do not change the PR scope, do not modify the default branch, and do not merge.

**Comment by jonathanhaaswriter** [comment]
@droid review

File: sdk/typescript/src/index.ts [hunk 1]
```diff
@@ -481,6 +518,36 @@ export class IntegrationClient {
     return this.client.listClaims(this.runtimeId, options);
   }
 
+  async graphNeighborhood(root: EntityRef | string, limit = 0): Promise<GraphNeighborhood> {
+    const rootUrn = typeof root === "string" ? root.trim() : root.urn.trim();
+    return this.client.getEntityNeighborhood(rootUrn, limit);
+  }
+
+  async graphLayering(roots: Array<EntityRef | string>, limit = 0): Promise<GraphLayering> {
+    const layering: GraphLayering = {};
```
[the remark this answers is not in the document]
Comment (jonathanhaaswriter) [reply sdk/typescript/src/index.ts]: Fixed in 043af05ca by creating the `graphLayering` map with `Object.create(null)`, preserving special keys like `__proto__`; type-check passed with `tsc --noEmit`.

**Comment by jonathanhaaswriter** [comment]
@droid review

Please re-review the SDK graph-layering surface for Jira posture deeply. This is the first SDK consumer of the bounded neighborhood query (PR 460), so contract drift is the main risk:

1. **Bounded read enforcement**: verify SDK helpers cannot accidentally request unbounded neighborhoods, that depth/fanout/total-node defaults are sensible, and that callers cannot bypass them through optional kwargs.
2. **Tenant/runtime scoping**: ensure both Python and TS helpers always pass the runtime/tenant context and that examples can’t be misused to read across tenants by passing the wrong runtime id.
3. **Error handling and graceful degradation**: confirm graph-API errors return well-typed, sanitized error payloads (no internal details leaked) and that examples continue to render claim data when the graph surface is unavailable.
4. **Output normalization**: verify nodes/relations returned to SDK callers are normalized identically across Python and TS, with stable ordering and consistent ID/URN handling, so downstream graph-summary helpers (PR 469/470) don’t see drift.
5. **Auth/SDK token plumbing**: re-check token handling for the new graph reads—no logging, scoped to the SDK runtime, and consistent with claim read/write helpers.
6. **Forward-compat**: confirm the graph-layering response shape stays stable for PR 469 (summarize layering), PR 471 (derive findings), and any future graph SDK helpers.
7. **Onboarding example UX**: ensure the Jira posture example renders both claims and graph layering in a way that makes failure modes (graph unavailable, empty neighborhood, errored fetch) obvious to integrators rather than silently empty.

**Comment by jonathanhaaswriter** [comment]
Closing this PR — its substantive feature work is already in `main` via the earlier consolidation rollups, and the unresolved factory-droid findings posted on this PR have been addressed in main via the follow-up PRs:

- #497 (`fix(github): drop non-cursor opaque from terminal audit checkpoints`)
- #499 (`fix: address post-merge feedback across pending stacks`)

PR #499 specifically addresses the findings posted on this stack:

- Reject malformed `root_urn` values in `GetEntityNeighborhood` (PR #460)
- Map `refuted` and `superseded` claim statuses to projection deletion (PR #463)
- Tenant-scope `ListClaims` reads (PR #464)
- Per-runtime serialization of `WriteClaims` (PR #474)
- Restore null-array normalization in `graph_summary` (PR #470)
- Surface blank-root errors from `graphLayering` (PR #468)

This branch has diverged ~50k lines from `main` and rebasing it cleanly is no longer practical; closing here keeps the open queue accurate. If anything from this stack still needs to land, please open a fresh PR cut from `main`.
